### DISCUSSION TITLE
Not able to send command to rest device service json resource from app service

### DISCUSSION THREAD

**Discussion by sudhamani-hcl** [edgexfoundry/community#228] [General]
Hi Team,

I am trying to send core command to rest device service from EdgeX app service.
I am using `CommandClient().IssueSetCommandByName()` API in EdgeX app service.

Rest device service is expecting  data in the format `map[string]map[string]string` .
For ex: `map[json:map[id:100 name:"xxx"]]`.
Here;
Resource type is `object type`
Resource name is `json` 

However `IssueSetCommandByName()` API is accepting `map[string]string` type data only. 
How can I send `map[json:map[id:100 name:"xxx"]]` data to json resource from app service?
Any help would be appreciated.

Thanks,
Sudhamani

**Comment by lenny-goodell** [comment]
You have to use `IssueSetCommandByNameWithObject` API on the command client.
[IssueSetCommandByNameWithObject](https://github.com/edgexfoundry/go-mod-core-contracts/blob/v3.1/clients/interfaces/command.go#L34)

**Comment by sudhamani-hcl** [comment]
Thanks @lenny-intel  it was helpful. Hence closing the discussion.
